### PR TITLE
Introduce PSR-7 `ResponseInterface` for Modules executions

### LINKED ISSUE #11033: Checkbox "Notification" is always empty
- [x] I have searched open and closed issues for duplicates

### Bug Description

Can not see, if i follow a contact or not. Can not follow contact

### Steps to Reproduce

1.  Search for an account
2.  Go to profile->contact setting
3.  check "Notify me on new acitivites"
4. Klick "Send"
5. Open "Contact settings" again

Actual Result:

Checkbox is not checkt, when reopen contact-setting or even the whole page.
No new Notifications possible.

Expected Result:

When i check "Notify me" and Send the form, i would expect to see the checkbox checked on reload.
When checked, i want to receive notifications about new postings/activities from this contact

### Platform Info

Friendica Version: Friendica 'Siberian Iris' 2021.12-dev - 1445

Friendica Source: 

PHP version: 7.4

SQL version: mariadb10.3/Debian

### PR DESCRIPTION
**Pull request by nupplaphil**
FollowUp #11002 

Split and delete `ModuleController`:
- $moduleName is part of the argument string => App\Arguments
- $isBackend boolean already part of App\Mode::isBackend()
- $module is now the direct return of App\Router::getModule()
- ModuleController::run() moved to BaseModule::run()

Introduce `Response` class and corresponding interfaces to create a standard PSR-7 compliant `ResponseInterface`
This particular interface (`ResponseInterface`) withis now returned after each `Module` call (including the `API` calls) and makes all `Module`s testable

Needs https://github.com/friendica/friendica-addons/pull/1209 as well for rendertime :-)

### CODE REVIEW

File: src/App.php [hunk 1]
```diff
@@ -703,20 +700,20 @@ public function runFrontend(App\ModuleController $module, App\Router $router, IM
 			$page['page_title'] = $moduleName;
 
 			if (!$this->mode->isInstall() && !$this->mode->has(App\Mode::MAINTENANCEDISABLED)) {
-				$module = new ModuleController('maintenance', new Maintenance());
+				$module = new Maintenance();
```
Comment (MrPetovan) [inline src/App.php]: Needs `L10n` instance as per your recent fixings.

File: src/App.php [hunk 2]
```diff
@@ -703,20 +700,20 @@ public function runFrontend(App\ModuleController $module, App\Router $router, IM
...
 			} else {
 				// determine the module class and save it to the module instance
 				// @todo there's an implicit dependency due SESSION::start(), so it has to be called here (yet)
-				$module = $module->determineClass($this->args, $router, $this->config, $dice);
+				$module = $router->getModule();
```
Comment (MrPetovan) [inline src/App.php:702]: What about legacy modules still in `mod/`?

File: src/App/Mode.php [hunk 3]
```diff
@@ -25,6 +25,7 @@
 use Friendica\Core\Config\ValueObject\Cache;
 use Friendica\Database\Database;
 use Friendica\Util\BasePath;
+use phpDocumentor\Reflection\Types\Static_;
```
Comment (MrPetovan) [inline src/App/Mode.php]: I'm sorry what?

File: src/App/Router.php [hunk 4]
```diff
@@ -83,6 +91,15 @@ class Router
 	/** @var ICanLock */
 	private $lock;
 
+	/**	@var Arguments */
```
Comment (MrPetovan) [inline src/App/Router.php]:
```suggestion
	/** @var Arguments */
```

File: src/App/Router.php [hunk 5]
```diff
@@ -91,14 +108,21 @@ class Router
 	 * @param string              $baseRoutesFilepath The path to a base routes file to leverage cache, can be empty
 	 * @param L10n                $l10n
 	 * @param ICanCache           $cache
+	 * @param ICanLock            $lock
```
Comment (MrPetovan) [inline src/App/Router.php:111]: This parameter doesn't exist.

**Review by MrPetovan — CHANGES_REQUESTED** [review]
I like all the deletions but so far this PR doesn't pass the checks.

File: src/App.php (hunk 2 above)
Comment (nupplaphil) [reply src/App.php:702]: It's now moved directly into the router. The router now didn't "just" find the right parameters & classes, it also creates the class it with the help of `Dice`, see the new `Router::getModule()` method

File: src/App/Mode.php (hunk 3 above)
Comment (nupplaphil) [reply src/App/Mode.php]: hehe, sometimes I start typing `stat..` and hit the "Tab" Button, and voliá, the IDE thinks importing `Static_` seems the right guess ;-)

Comment (nupplaphil) [reply src/App/Mode.php]: normally I don't forget to remove it again .. not this time :D

File: src/App/Router.php (hunk 5 above)
Comment (nupplaphil) [reply src/App/Router.php:111]: of course it does o_O see https://github.com/friendica/friendica/pull/11005/files/a7a05633d8226edb76eb9a9f0bce184db82bd4a3#diff-7f7aabf2d5c355cac62f512a0d5387e00dfcb00b27e6dcb39b3b90ff028d05dfR122

**Comment by nupplaphil** [comment]
Another big PR, but we're on a good way to test any module we want without mocking/spying/..

Tests are not in my focus yet, I'll check them later :D

-> it's now running at https://friendica.philipp.info
-> I'll check for any errors (post/preview/delete still working ;-) )

**Comment by nupplaphil** [comment]
Ok, I think I leave it for now...
... Testing the whole branch for at least another day 

At least workers are running fine so far and I can post/update/delete , see every profile, can edit permissions, ...

But **why** did I change that much:
See https://github.com/friendica/friendica/pull/11005/commits/9494e2e4b389c8c2f69304bb3782c71fdbd99537

We can now do tests for complete modules :-)
With the help of `Response` every module now returns an interface with:
- `getContent()` - A mixed content, set by the Module (the core output)
- `getType()` - The type of content (HTML, RSS, ...)
- `getHeaders()` - An array with headers, we **would** set

Currently, almost every Module directly uses `header()` and `System::*()` calls to exit directly, but we can now replace this calls step by step directly with new method calls `$this->response->setHeader(string $header, ?string $key)` and `$this->response->addContent($content)` and add a test for every transformed module if we want to ;-)

**Comment by MrPetovan** [comment]
While we're at it, can we use [PSR-7 compatible messages](https://www.php-fig.org/psr/psr-7/)? We already depend on the Guzzle implementation, so we can directly use `GuzzleHttp\Psr7\Response` without needing an additional dependency.

**Review by MrPetovan — APPROVED** [review]

**Comment by nupplaphil** [comment]
No errors so far in my logs for a week ..

--> I rebased the changes of this week into this branch and start a smoke test again :)

**Comment by nupplaphil** [comment]
The next PR will cover the `HTTPInputData` class to make the whole API changes from @annando testable again (I tried it to add it at this PR, but as said I need to change the HTTPInput behavior) :-)

File: src/Module/Contact/Profile.php [hunk 6]
```diff
@@ -355,7 +355,7 @@ protected function content(array $request = []): string
 			'$archived'                  => ($contact['archive'] ? $this->t('Currently archived') : ''),
 			'$pending'                   => ($contact['pending'] ? $this->t('Awaiting connection acknowledge') : ''),
 			'$hidden'                    => ['hidden', $this->t('Hide this contact from others'), $localRelationship->hidden, $this->t('Replies/likes to your public posts <strong>may</strong> still be visible')],
-			'$notify'                    => ['notify', $this->t('Notification for new posts'), ($contact['notify_new_posts'] == 1), $this->t('Send a notification of every new post of this contact')],
+			'notify_new_posts'           => ['notify_new_posts', $this->t('Notification for new posts'), ($localRelationship->notifyNewPosts), $this->t('Send a notification of every new post of this contact')],
```
Comment (MrPetovan) [inline src/Module/Contact/Profile.php]: Please respect the `$` notation for template variables, both work but every other key in the code is prefixed with a dollar sign.

**Review by MrPetovan — CHANGES_REQUESTED** [review]

**Comment by nupplaphil** [comment]
so, now I hopefully fixed everything.
Note to myself: Dice doesn't pass arguments from `$dice->create(Class::class, $arguments)` to any other class except the initial class. Means that `Tos` didn't got the original parameters when adding as implicit dependency at the constructor :-/

**Review by MrPetovan — APPROVED** [review]

**Pull request merged**